lib/src/linter.dart

```diff
@@ -325,9 +331,11 @@ class SourceLinter implements DartLinter, AnalysisErrorListener {
 
   @override
   Iterable<AnalysisErrorInfo> lintPubspecSource(
-      {String contents, String sourceUrl}) {
+      {String contents, String sourcePath}) {
     var results = <AnalysisErrorInfo>[];
 
+    Uri sourceUrl = sourcePath == null ? null : p.toUri(sourcePath);
```

Comment:
Nit: maybe call it `sourceUri`?  It's confusing to have its type be `Uri` but to call it a "Url".